bot.py

```diff
@@ -97,17 +101,15 @@ async def dinkster(self, message, command_content):
         for channel in await message.guild.fetch_channels():
             if isinstance(channel, discord.VoiceChannel):
                 voice_client = await channel.connect()
-                audio_source = await discord.FFmpegOpusAudio.from_probe(
-                    "Dinkster.ogg"
-                )
+                audio_source = await discord.FFmpegOpusAudio.from_probe("Dinkster.ogg")
                 voice_client.play(audio_source)
                 await asyncio.sleep(10)
                 await voice_client.disconnect()
 
     async def play(self, message, command_content):
         video_metadata = None
 
-        #This looks pretty ugly
+        # This looks pretty ugly
         try:
```

Comment:
How about using validators.url [1] instead of the try-except? Something like:

```
if validators.url(command_content):
  video_metadata = pafy.new(command_content)
else:
  search_result = VideosSearch(command_content).result()
  video_metadata = pafy.new(search_result["result"][0]["id"])
```

[1] https://validators.readthedocs.io/en/latest/#validators.url.url

Reply:
Can pafy handle e.g. soundcloud links? If not, @gardarandri's code will raise an error. I would like this more explicit check nonetheless; the intent of the code is clearer, and we can log actual errors if they appear.

Reply:
yes